pilot/pkg/serviceregistry/serviceentry/store.go

```diff
@@ -70,11 +70,11 @@ func (s *serviceInstancesStore) deleteInstanceKeys(key configKey, instances []*m
 		if len(s.instances[ikey]) == 0 {
 			delete(s.instances, ikey)
 		}
-		delete(s.ip2instance, i.Endpoint.Address)
+		delete(s.ip2instance, i.Endpoint.Key())
```

Comment:
I suspect this usage will break if we have an endpoint change from having 1 IP to 2. Since we only include the first IP, we may miss it and leak (will impact behavior, not just memory leak).

We have no test coverage here.

Reply:
Hi @howardjohn, I understand your worries, so I added some other tests functions for `addInstances` and `deleteInstanceKeys` in `store.go`, the important changes is in `store_test.go`.

Reply:
@zhlsunshine : I think what John says here is if there is an update to endpoint where the IP changes from single to multiple, the new logic will not work. Could you please add some unit tests to "updateInstances()" by updating the ip address of endpoint, in store.go and see how it behaves?

Reply:
Hi @kfaseela & @howardjohn ,  in fact these logic has already been included in `TestServiceInstancesStore`` and `TestServiceInstancesForDnsRoundRobinLB` in `store_test.go`, besides that, some other unit tests for adding or deleting an instance already cover these logic.

Reply:
Hi @kfaseela, I added a new test case `TestUpdateInstances` which contains 2 cases: the single address to multiple address and multiple address to single address.